shopping_mall/backend/ai/seed_rag.py

```
@@ -106,12 +179,93 @@ def chunk_by_sections(text: str, source: str) -> list[dict]:
+    if not articles:
+        return []
+
+    # 장·조 모든 분할점을 위치순으로 정렬
+    splits: list[dict] = []
+    for m in chapters:
+        splits.append({
+            "type": "chapter",
+            "pos": m.start(),
+            "num": m.group(1),
+            "title": m.group(2).strip(),
+        })
```

Ruff (0.15.10)

[warning] Use a list comprehension to create a transformed list

(PERF401)